src/Npgsql/Internal/Composites/Metadata/CompositeBuilder.cs

```diff
@@ -87,16 +90,14 @@ protected override void Construct()
         }
     }
 
-    protected override void SetField<TValue>(TValue value)
+    public void Reset()
     {
-        if (_boxedInstance is null)
-            ThrowHelper.ThrowInvalidOperationException("Not constructed yet, or no more fields were expected.");
-
-        var currentField = _currentField;
-        var fields = compositeInfo.Fields;
-        if (currentField > fields.Count - 1)
-            ThrowHelper.ThrowIndexOutOfRangeException($"Cannot set field {value} at position {currentField} - all fields have already been set");
-
-        ((CompositeFieldInfo<TValue>)fields[currentField]).Set(_boxedInstance, value);
+        _instance = default!;
+        _boxedInstance = null;
+        _currentField = 0;
+        foreach (var box in _tempBoxes)
+            box.Clear();
     }
+
+    public void Dispose() { }
```

Comment:
Is there any reason to even have `Dispose` anymore given that it doesn't do anything and the class is sealed?

Reply:
Yeah that's fine it's also internal code so no need to keep it around.